plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueCatalogAccessOperations.java

```diff
@@ -100,26 +81,41 @@ protected QueryRunner createQueryRunner()
         File tmp = Files.createTempDirectory("test_iceberg").toFile();
         DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(testSession).build();
 
-        AtomicReference<GlueMetastoreStats> glueStatsReference = new AtomicReference<>();
+        spanExporter = closeAfterClass(InMemorySpanExporter.create());
+
+        SdkTracerProvider tracerProvider = SdkTracerProvider.builder()
+                .addSpanProcessor(SimpleSpanProcessor.create(spanExporter))
+                .build();
+
+        OpenTelemetry openTelemetry = OpenTelemetrySdk.builder()
+                .setTracerProvider(tracerProvider)
+                .build();
+
         queryRunner.installPlugin(new TestingIcebergPlugin(
                 Optional.empty(),
                 Optional.empty(),
-                new StealStatsModule(glueStatsReference)));
+                binder -> {
+                    Multibinder<RequestHandler2> requestHandlers = newSetBinder(binder, RequestHandler2.class, ForGlueHiveMetastore.class);
```

Comment:
This is not quite correct.
If we do this way, we end up with having 2 `TracingRequestHandler` instances defined in `HiveGlueClientProvider`.

Please consider passing the `OpenTelemetry` instance in the same way it was done in 

https://github.com/trinodb/trino/blob/6849365d5a492a38de33404b3881959642c21cb3/plugin/trino-hive/src/main/java/io/trino/plugin/hive/InternalHiveConnectorFactory.java#L131

Reply:
That's intentional. I'm trying to avoid changing InternalHiveConnectorFactory.

Reply:
> I'm trying to avoid changing InternalIcebergConnectorFactory.

Please detail why do you avoid changing `InternalIcebergConnectorFactory`. What could be the side effects of this operation?